.github/workflows/run-tests.yml

```diff
@@ -122,22 +122,19 @@ jobs:
                      --env PUSH_TO_CDASH=true \
                      --env WITH_3D=${WITH_3D} \
                      qgis/qgis3-build-deps:${DOCKER_TAG} \
-                     /root/QGIS/.docker/docker-qgis-build.sh
-          docker commit qgis_container qgis_image
+                     /bin/bash -c "/root/QGIS/.docker/docker-qgis-build.sh; ctest -V -S /root/QGIS/.ci/config_build.ctest"
 
-      - name: Push image
-        if: github.event_name == 'pull_request' && contains( github.event.pull_request.labels.*.name, 'Push Docker Image')
-        env:
-          GITHUB_PR_NUMBER: ${{github.event.number}}
-        run: |
-          docker login -u="${DOCKER_USERNAME}" -p="${DOCKER_PASSWORD}"
-          docker commit qgis_container qgis/qgis-pr-build:PR${GITHUB_PR_NUMBER}
-          docker push qgis/qgis-pr-build:PR${GITHUB_PR_NUMBER}
+      - name: Push artifact
+        id: push_artifact
+        if: ${{ matrix.ubuntu-base == env.DEFAULT_UBUNTU_BASE }}
```

Comment:
could you maybe introduce a variable to define if we run the tests or no, ideally in the strategy?

Reply:
Done 72e66ee